### PR TITLE
feat(nested-menu): support nested menu

### PR DESCRIPTION
**Pull request by zverbeta**

### CODE REVIEW

**Comment by Kaskyi** [comment]
Alternatively (No worries about backward compatibility), you can create a standalone component and provide it through the public configuration(admin/config) via `sidebar: {layout:[{selector: "@xm-ngx/componets/your-menu-for-sidebar"}]}`

**Comment by zverbeta** [comment]
@Kaskyi From my point of view, maintaining one type of menu is much easier than several.

File: packages/components/menu/flat-menu.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { JavascriptCode } from '@xm-ngx/shared/interfaces';
+import { ContextService } from '../../../src/app/shared';
+import { transpilingForIE } from '@xm-ngx/json-scheme-form';
+import { Dashboard } from '@xm-ngx/dashboard';
+import { MenuItem } from '@xm-ngx/components/menu/menu.interface';
+import { XmEntitySpec } from '@xm-ngx/entity';
+import * as _ from 'lodash';
+
+function checkCondition(item: { config?: { condition?: JavascriptCode } }, contextService: ContextService): boolean {
+    // If configurator do not provide configs, return true
```
Comment (oscaryald) [inline packages/components/menu/flat-menu.ts:10]: unnecessary comment

File: packages/components/menu/menu.component.ts [hunk 2]
```diff
@@ -164,14 +44,16 @@ export class MenuComponent implements OnInit, OnDestroy {
     ) {
     }
 
+    public hasChild = (_: number, node: any): boolean => node.isLink == null ? (!!node.children && node.children.length > 0) : !node.isLink;
+
     public ngOnInit(): void {
         const dashboards$ = this.dashboardService.dashboards$().pipe(
             startWith([]),
-            takeUntilOnDestroy(this),
+            takeUntil(this.unsubscribe),
```
Comment (oscaryald) [inline packages/components/menu/menu.component.ts:52]: in project we using takeUntilOnDestroy(this) instead takeUntil(this.unsubscribe)

File: packages/components/menu/menu.component.ts [hunk 3]
```diff
@@ -184,65 +66,91 @@ export class MenuComponent implements OnInit, OnDestroy {
...
-            map(applicationsToCategory),
+            map((dashboards) => applicationsToCategory(dashboards)),
         );
 
         const default$ = this.uiConfigService.config$().pipe(
             take(1),
-            map(i => i?.sidebar?.hideAdminConsole ? [] : DEFAULT_MENU_LIST),
+            map(i => i?.sidebar?.hideAdminConsole ? [] : getDefaultMenuList()),
             shareReplay(1),
+            takeUntil(this.unsubscribe),
```
Comment (oscaryald) [inline packages/components/menu/menu.component.ts:79]: unnecessary, because already has take(1)

File: packages/components/menu/menu.component.ts [hunk 4]
```diff
@@ -184,65 +66,91 @@ export class MenuComponent implements OnInit, OnDestroy {
...
         );
 
-        this.categories$ = combineLatest([dashboards$, applications$, default$]).pipe(
-            takeUntilOnDestroy(this),
-            map(([a, b, c]) => [...a, ...b, ...c]),
+        this.categories$ = combineLatest([ dashboards$, applications$, default$ ]).pipe(
+            map(([ a, b, c ]) => [ ...a, ...b, ...c ]),
             shareReplay(1),
+            takeUntil(this.unsubscribe),
```
Comment (oscaryald) [inline packages/components/menu/menu.component.ts:85]: unsubscribe should be on the top of chain

File: packages/components/menu/menu.component.ts [hunk 5]
```diff
@@ -184,65 +66,91 @@ export class MenuComponent implements OnInit, OnDestroy {
...
-            } else {
-                _.forEach(category.children, (item) => {
-                    activateCategory(category, item.url);
-                });
-            }
-        });
+    public toggle(node: MenuItem, evt?: Event): void {
+        evt?.preventDefault();
+
+        // If node is expand, collapse it and exit from toggle function
```
Comment (oscaryald) [inline packages/components/menu/menu.component.ts:134]: pls remove unnecessary comments

**Review by oscaryald — APPROVED** [review]

**Review by Sumragen — DISMISSED** [review]

**Comment by Sumragen** [comment]
@Kaskyi Could you please review

**Review by Sumragen — DISMISSED** [review]

**Review by Kaskyi — DISMISSED** [review]

**Review by Kaskyi — DISMISSED** [review]

**Review by Kaskyi — APPROVED** [review]

**Review by Sumragen — APPROVED** [review]

**Pull request merged**